pages/PreviewPage/[searchID].js

```diff
@@ -5,32 +5,55 @@ import Header from "../../components/Header";
 import PreviewPicture from "../../components/PreviewPicture";
 import { saveAs } from "file-saver";
 import { css } from "styled-components";
+import useSWR from "swr";
 
-const imageSrc = "https://i.imgur.com/8nUXl6s.jpeg/";
-const imageName = "8nUXl6s.jpeg";
+const fetcher = (url) => fetch(url).then((res) => res.json());
 
 export default function PreviewPage() {
   const router = useRouter();
-  const imageID = "8nUXl6s";
+  const { searchID } = router.query;
+  const option = router.query.option;
 
-  function handleVariations() {
-    router.push(`/Variations/${imageID}`);
+  const {
+    data: shirts,
+    isLoading,
+    error,
+  } = useSWR(`/api/ChooseVariation/${searchID}`, fetcher);
```

Comment:
Again, we should consider that `searchID` might be undefined on the first render and we should handle that. (see previous comment).